### ISSUE TITLE
"find . -name '*.sh' | xargs shellcheck" exited with 123.

### ISSUE THREAD

**Issue by jeff1evesque** [koalaman/shellcheck#632]
My [`.travis.yml`](https://github.com/jeff1evesque/machine-learning/blob/5a146f51ec4ad28f6560de8faef2605dc2a695e0/.travis.yml) implementation:

```
sudo: required

language: python

## allows 'docker' command
services:
  - docker

## build docker container for unit testing.
before_install:
  - docker ps -a
  - docker build -t ubuntu/trusty .

## install packages for linting
#
#  @trusty-backports, allows additional packages (i.e. shellcheck) to be installed
#
#  Note: unit testing is performed within the docker container, which is
#        defined from the Dockerfile.
install:
  - sudo add-apt-repository "deb http://archive.ubuntu.com/ubuntu/ trusty-backports restricted main universe"
  - sudo apt-get install shellcheck

## implement linting
#
#  @puppet-lint, the following issue needs to be checked if it has been
#      resolved, so the '--no-ensure_first_param-check' abusive flag
#      implementation (below) can be removed:
#
#      - https://github.com/rodjek/puppet-lint/issues/410
#
#  Note: unit testing is performed within the docker container, which is
#        defined from the Dockerfile.
script:
  - find . -type f -exec grep -l '^#!/bin/bash' {} + | xargs shellcheck
  - find . -name '*.sh' | xargs shellcheck

```

generates the following [traceback](https://travis-ci.org/jeff1evesque/machine-learning/builds/118008592#L1068-L1079), via our travis ci implementation:

```bash
...
$ find . -name '*.sh' | xargs shellcheck

No files specified.

Usage: shellcheck [OPTIONS...] FILES...

  -f FORMAT         --format=FORMAT          output format

  -e CODE1,CODE2..  --exclude=CODE1,CODE2..  exclude types of warnings

  -s SHELLNAME      --shell=SHELLNAME        Specify dialect (bash,sh,ksh,zsh)

  -V                --version                Print version information

The command "find . -name '*.sh' | xargs shellcheck" exited with 123.
...
```

**Comment by jeff1evesque** [comment]
The following should be the working directory for the `shellcheck` implementation:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ ls
app.py  contributing.md  interface   puppet       settings.pyc  Vagrantfile
brain   Dockerfile       license.md  README.md    src
build   __init__.py      log         settings.py  test
```

**Comment by koalaman** [comment]
You have no `.sh` files so xargs calls the program once with zero arguments.

Call shellcheck from a find `-exec` to both avoid calling it without input and to better handle special characters in filenames

**Comment by jeff1evesque** [comment]
When I run the `find` command from the directory containing my `.travis.yml`, I notice there are multiple `.sh` files:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ find . -name '*.sh'
./build/scikit-learn/continuous_integration/install.sh
./build/scikit-learn/continuous_integration/test_script.sh
./build/scikit-learn/sklearn/externals/copy_joblib.sh
./puppet/environment/development/modules_contrib/git/files/subtree/bash_completi
on.sh
./puppet/environment/development/modules_contrib/vcsrepo/spec/acceptance/files/c
reate_git_repo.sh
```

**Comment by jeff1evesque** [comment]
@koalaman, ah I see. Those files exist, after puppet has provisioned my environment.  The linting, which checks for `*.sh` files, doesn't have the provisioned environment.  But, in all honesty, those files are from some other project codebase, and I shouldn't be linting their code, anyways.

I used to have a shell file (which update puppet on the guest vagrant vm), which I removed.  I forgot this, which is why I am getting the error.  I will look into using the `-exec` implementation, as you suggest.  Thank you!
